### PR TITLE
Make test functions isolated

### PR DESCRIPTION
**Pull request by TharmiganK**
Part of: https://github.com/ballerina-platform/ballerina-library/issues/6120

This is to run the tests in parallel with the new parallel test feature

## Examples

N/A

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

**Review by Thevakumar-Luheerathan — DISMISSED** [review]

File: ballerina/tests/constriant_error_message_test.bal [hunk 1]
```diff
@@ -161,8 +161,8 @@ Trainee TRAINEE = {
 };
 
 @test:Config {}
-function testErrorMessageInRecordTypePositive() {
-    Trainee trainee = TRAINEE.clone();
+isolated function testErrorMessageInRecordTypePositive() returns error? {
+    Trainee trainee = check TRAINEE.cloneWithType();
```
Comment (SasinduDilshara) [inline ballerina/tests/constriant_error_message_test.bal]: Do we need to clone this record, because it is readonly?

Comment (TharmiganK) [reply ballerina/tests/constriant_error_message_test.bal]: For this test we do not need. I think I just replace all the places where I use `TRAINEE.clone()`. Will change this 👍🏼

Comment (TharmiganK) [inline ballerina/tests/constriant_error_message_test.bal]:
```suggestion
```

File: ballerina/tests/constriant_error_message_test.bal [hunk 2]
```diff
@@ -161,8 +161,8 @@ Trainee TRAINEE = {
...
     Trainee|error validation = validate(trainee);
```
Comment (TharmiganK) [inline ballerina/tests/constriant_error_message_test.bal]:
```suggestion
    Trainee|error validation = validate(TRAINEE);
```

File: ballerina/tests/constriant_error_message_test.bal (hunk 1 above)
Comment (TharmiganK) [reply ballerina/tests/constriant_error_message_test.bal]: Fixed via [175d48f](https://github.com/ballerina-platform/module-ballerina-constraint/pull/105/commits/175d48fda6703a8f238cee5bb78c2f971106953a)

File: ballerina/tests/constriant_error_message_test.bal [hunk 3]
```diff
@@ -161,9 +161,8 @@ Trainee TRAINEE = {
 };
 
 @test:Config {}
-function testErrorMessageInRecordTypePositive() {
-    Trainee trainee = TRAINEE.clone();
-    Trainee|error validation = validate(trainee);
+isolated function testErrorMessageInRecordTypePositive() returns error? {
```
Comment (TharmiganK) [inline ballerina/tests/constriant_error_message_test.bal]:
```suggestion
isolated function testErrorMessageInRecordTypePositive() {
```

**Review by SasinduDilshara — APPROVED** [review]

**Pull request merged**